### ISSUE TITLE
Monitoring for job failures

### ISSUE THREAD

**Issue by skofman1** [NuGet/NuGetGallery#3103]
This task is opened after this issue was discovered: https://github.com/NuGet/NuGetGallery/issues/3102

Failures in HandlePackageEdit go unnoticed. We need a better mechanism to handle failed requests and to monitor them.

**Comment by yishaigalatzer** [comment]
The main use case here is to unblock further package uploads/edits from happening

**Comment by joelverhagen** [comment]
What's the current detection mechanism? Just logs and user reports on edits not working?

**Comment by skofman1** [comment]
This is a duplicate of an in-progress work item: https://github.com/NuGet/Engineering/issues/2891
